src/core/syntax.ml

```diff
@@ -49,7 +49,7 @@ and p_term_aux =
   (** Abstraction over several variables. *)
   | P_Prod of p_arg list * p_term
   (** Product over several variables. *)
-  | P_LLet of ident * p_arg list * p_term * p_term
+  | P_LLet of ident * p_arg list * p_term * p_type option * p_term
```

Comment:
I would prefer to have the optional type first like in P_definition and like it is written:
```
P_LLet of ident * p_arg list * p_type option * p_term * p_term 
```

Reply:
Btw, don't we need to fix syntax.bnf now?

Reply:
The type argument here is the `T` in `let x := t : T in u` so I think it is in the right place.

Reply:
Actually no, you are write @fblanqui! I misread!

Reply:
The syntax file has already been modified